safe_transaction_service/account_abstraction/serializers.py

```diff
@@ -93,6 +78,39 @@ def _validate_signature(
             safe_signatures.append(safe_signature)
         return safe_signatures
 
+
+class SafeOperationSerializer(
+    SafeOperationSignatureValidatorMixin, serializers.Serializer
+):
+    nonce = serializers.IntegerField(min_value=0)
+    init_code = eth_serializers.HexadecimalField(allow_null=True)
+    call_data = eth_serializers.HexadecimalField(allow_null=True)
+    call_gas_limit = serializers.IntegerField(min_value=0)
```

Comment:
also this field is uint256.  Every integer field is uint256 type in the ERC4337 specification, do you think that we could have an overflow?

Reply:
In python `int` type is unbounded

Reply:
Ok, but other clients could have issues thinking that is int and receiving bigger numbers.

Reply:
Bigint is now native to js and other languages, so I expect js to take care of that. Just in case, @dasanra , WDYT?

Reply:
I think this can be more of an issue with the default parsing that is provided. JSON doesn't set any limits on `Number` but some frameworks might map a `Number` to lower bounded integer values causing an overflow. This is not the case for some implementations though. For example, JS has no issues with it:

```javascript
JSON.parse("{\"number\": 115792089237316195423570985008687907853269984665640564039457584007913129639936}")
{ number: 1.157920892373162e+77 }
```

I would say that the most friendly option towards the consumers of the API might actually be representing this as a `String` which moves the boundary validation to the consumers.

Reply:
Some extra information that might help deciding between one or the other:

> In fact, JSON allows number literals that are arbitrarily long; they just cannot be parsed to full precision in JavaScript. If you are communicating with another program in a language that supports longer integers (such as 64-bit integers), and you want to transmit the BigInt as a JSON number instead of a JSON string, see [Lossless number serialization](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON#using_json_numbers).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json

Reply:
Maybe we should deal with this in another issue, as the current existing endpoints are using Integers

Reply:
For some reason I'm subscribed to this issue and stumbled on this comment.

We had similar issues at CoW and came to the conclusion: big integers should be serialized to JSON strings over the wire.

Just to dump what we found during our investigations at the time (and how I remember it): JSON standard only specifies how numeric literals are [parsed](https://www.json.org), but does not impose any restrictions on the actual representation of those values. The actual underlying representation of numbers is left up to the parser. Some do exclusively IEEE 754 doubles, some do a combination of doubles and 64-bit integers (`serde` in Rust this this AFAIR), etc. Notably the `JSON.parse` method part of the ECMA script standard represents all numerical values with doubles by default.

Note that there are, in the ECMA script standard, ways to correctly parse these numbers ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#the_reviver_parameter)), but for compatibility purposes, strings are just easier to use for clients and typically suffer less from compatibility issues in ecosystems with simpler JSON parsers.

Anyway, unsolicited drive-by comment, Sorry :see_no_evil:.

Reply:
> Note that there are, in the ECMA script standard, ways to correctly parse these numbers ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#the_reviver_parameter)), but for compatibility purposes, strings are just easier to use for clients and typically suffer less from compatibility issues in ecosystems with simpler JSON parsers.

This has been my experience as well working with multiple parsers on different stack. Each parser sets constraints or fallback logic differently when it comes to trying to map the JSON value to a native type representation.

Given the parser's responsibility in all of this I'm of the opinion that a `String` is the value is most "compatible" and client-friendly format for big numbers.

Reply:
Ok, agree with going for CharField. But I would prefer to do it in a different PR and do the same for the existing 4337 endpoints